### PR TITLE
Changes required to bump Android SDK to 30

### PR DESCRIPTION
**Pull request by makingthematrix**
For now I made changes only to make the app compile again.
We should talk about the new way to set soft input mode (what are window insets? anyone?) and about the new way to configure AndroidCamera2.
#### APK
[Download build #2867](http://10.10.124.11:8080/job/Pull%20Request%20Builder/2867/artifact/build/artifact/wire-dev-PR3049-2867.apk)

### CODE REVIEW

**Comment by makingthematrix** [comment]
Here's an example how to use insets, but I'm not sure how it translates to our case:
https://github.com/StylingAndroid/WindowInsets/blob/Part1/app/src/main/java/com/stylingandroid/windowinsets/MainActivity.kt

File: app/src/main/kotlin/com/waz/zclient/feature/settings/account/editphonenumber/EditPhoneNumberActivity.kt [hunk 1]
```diff
@@ -11,7 +11,7 @@ import kotlinx.android.synthetic.main.activity_edit_phone.*
 class EditPhoneNumberActivity : AppCompatActivity(R.layout.activity_edit_phone) {
 
     private val phoneNumber: String by lazy {
-        intent.getStringExtra(CURRENT_PHONE_NUMBER_KEY)
+        intent.getStringExtra(CURRENT_PHONE_NUMBER_KEY) ?: ""
```
Comment (BradleyWire) [inline app/src/main/kotlin/com/waz/zclient/feature/settings/account/editphonenumber/EditPhoneNumberActivity.kt:14]: `?: String.empty() `

**Review by BradleyWire — APPROVED** [review]

**Comment by BradleyWire** [comment]
build again

**Review by android10 — APPROVED** [review]
:+1:

**Review by mejdoo — APPROVED** [review]

**Comment by mejdoo** [comment]
@BradleyWire I think we shouldn't merge that before checking the side effects of targeting sdk 30. @makingthematrix added DO NOT MERGE label to it. Maybe it's safer for now to secure target sdk 29 for the next release. 

@gizemb @BradleyWire any thoughts about that?

**Comment by mejdoo** [comment]
I added an item for discussion in the next collective.

**Comment by android10** [comment]
@mejdoo @gizemb @BradleyWire I think it is fair enough to discuss. I have not seen the label and I approved the PR.

**Pull request merged**